tests/lib/rules/array-bracket-spacing.js

```diff
@@ -159,8 +174,11 @@ ruleTester.run("array-bracket-spacing", rule, {
         { code: "var foo = [1, {'bar': 'baz'}, 5];", options: ["never"] },
         { code: "var foo = [{'bar': 'baz'}, 1,  5];", options: ["never"] },
         { code: "var foo = [1, 5, {'bar': 'baz'}];", options: ["never"] },
-        { code: "var obj = {'foo': [1, 2]}", options: ["never"] }
+        { code: "var obj = {'foo': [1, 2]}", options: ["never"] },
 
+        // destructuring with type annotation
+        { code: "([ 1, 1 ]: Array<any>)", options: ["always"], parserOptions: { ecmaVersion: 6 }, parser: parser("flow-destructuring-1") },
+        { code: "([1, 1]: Array< any >)", options: ["never"], parserOptions: { ecmaVersion: 6 }, parser: parser("flow-destructuring-2") },
```

Comment:
Is this valid syntax?
As far as I know, the array patterns of destructuring cannot exist independently.

- It needs the initializer of variable declarations: e.g. `let [a] = b`
- It needs to be at LHS of assignments: e.g. `([a] = b)`
- It needs to be in parameters: e.g. `([a]) => a`

Also, those AST looks `ArrayExpression`, not `ArrayPattern`. Those are not destructuring.